[FLINK-25129][docs] Improvements to the table-planner-loader related docs (#18812)
apache/flink: docs/content/docs/dev/configuration/connector.md

# Connectors and Formats

Flink can read from and write to various external systems via connectors and define the format in 
which to store the data.
Flink can read from and write to various external systems via connectors and use the format of your choice
in order to read/write data from/into records.

> The way that information is serialized is represented in the external system and that system needs

Reviewer: Imho, we should also keep this paragraph.

Reviewer: Not insisting, let's also hear other's opinions, but I'd like this 2 sentence extra explanation about the formats, even if it's explained in detail in the provided link.

Reviewer 2: My proposal would be:

```
Flink applications can read from and write to various external systems via connectors. 
It supports multiple formats in order to encode and decode data to match Flink's data structures.
```

Author: Updated as you suggested @MartijnVisser

In order to use connectors and formats, you need to make sure Flink has access to the artifacts implementing them. 

> For each connector supported by the Flink community, we publish on [Maven Central](https://search.maven.org) two artifacts:

Reviewer 3: ```suggestion
For each connector supported by the Flink community, we publish two artifacts on [Maven Central](https://search.maven.org):
```

> * `flink-connector-<NAME>` which is a thin JAR including only the connector code, but excluding eventual 3rd party dependencies

Reviewer 3: ```suggestion
* `flink-connector-<NAME>` which is a thin JAR including only the connector code, but excluding eventual third-party dependencies
```

> * `flink-sql-connector-<NAME>` which is an uber JAR ready to use with all the connector 3rd party dependencies.

Reviewer 3: ```suggestion
* `flink-sql-connector-<NAME>` which is an uber/fat JAR ready to use with all the connector third-party dependencies.
```

> The same applies for formats as well. Also note that some connectors, because they don't require 3rd party dependencies,
> may not have a corresponding `flink-sql-connector-<NAME>` artifact.

Reviewer 3: ```suggestion
The same applies for formats as well. Note that some connectors may not have a corresponding `flink-sql-connector-<NAME>` artifact because they do not require third-party dependencies,.
```

{{< hint info >}}

> The uber JARs are supported mostly for being used in conjunction with [SQL client]({{< ref "docs/dev/table/sqlClient" >}}),
> but you can also use them in any DataStream/Table job.

Reviewer 3: ```suggestion
The uber/fat JARs are supported mostly for being used in conjunction with the [SQL client]({{< ref "docs/dev/table/sqlClient" >}}),
but you can also use them in any DataStream/Table job.
```
